plugins/woocommerce/src/Utilities/NumberUtil.php

```diff
@@ -49,4 +53,85 @@ public static function array_sum( array $arr ): float {
 
 		return array_sum( $sanitized_array );
 	}
+
+	/**
+	 * Sanitize a cost value based on the current locale decimal and thousand separators.
+	 *
+	 * @param string $value               The value to sanitize.
+	 * @return string                     The sanitized value.
+	 * @throws \InvalidArgumentException If the value is not a valid numeric value.
+	 */
+	public static function sanitize_cost_in_current_locale( $value ): string {
+		$value                      = is_null( $value ) ? '' : $value;
+		$value                      = wp_kses_post( trim( wp_unslash( $value ) ) );
+		$currency_symbol_encoded    = get_woocommerce_currency_symbol();
+		$currency_symbol_variations = array( $currency_symbol_encoded, wp_kses_normalize_entities( $currency_symbol_encoded ), html_entity_decode( $currency_symbol_encoded ) );
+
+		$value = str_replace( $currency_symbol_variations, '', $value );
+
+		// Count the number of decimal points.
+		$decimal_point_count = substr_count( $value, '.' );
```

Comment:
> @waclawjacek does the [commit](https://github.com/woocommerce/woocommerce/pull/54181/commits/5eb5e149fdd1dace6658749dcfb3d4646d917e2c) I pushed look ok to you? I'd like to get this into a 9.7.0 CFE soon

@rjchow I'm not entirely sure here. If `.` has been defined as the thousand separator, wouldn't we be making an incorrect assumption by hardcoding the decimal pointer?
```suggestion
		$decimal_point_count = substr_count( $value, wc_get_price_decimal_separator() );
```

Reply:
This is hardcoded because we expect the frontend to produce standard PHP numbers with only `.` for decimal separators, after this PR is merged. This is a way of cleaning up the database going forward so that we only have PHP numbers in the database instead of locale-specific numbers